common/src/types/v0/store/replica.rs

```diff
@@ -119,10 +207,20 @@ impl ResourceUuid for ReplicaSpec {
 
 impl From<ReplicaSpec> for models::ReplicaSpec {
     fn from(src: ReplicaSpec) -> Self {
-        Self::new(
+        let pool_id = match src.pool.clone() {
+            PoolRef::Named(id) => id,
+            PoolRef::Uuid(id, _) => id,
+        };
+        let pool_uuid = match src.pool {
+            PoolRef::Named(_) => None,
+            PoolRef::Uuid(_, uuid) => Some(uuid),
+        };
+        Self::new_all(
             src.managed,
+            None,
             src.owners,
-            src.pool,
+            pool_id,
+            pool_uuid.map(|uuid| openapi::apis::Uuid::try_from(uuid).unwrap()),
```

Comment:
we should not unwrap here, if need need we can use a TryFrom.
but in this case I don't think we need to, I think we can simply do pool_uuid.into_opt() here?